swagger.yaml

```diff
@@ -0,0 +1,226 @@
+info:
+  title: "Tweet API"
+  description: "This provides an API for creating a message (post), reading a message (get), editing a message (put), delete a message (delete)."
+  version: "1.0"
+  license:
+    name: GNU Affero General Public License Version 3
+    url: http://www.gnu.org/licenses/agpl-3.0.en.html
+externalDocs:
+  description: GitHub Repository
+  url: https://github.com/osu-mist/tweet-api
+host: localhost
+schemes:
+  - "https"
+basePath: /api/v1
+consumes:
+  - application/json
+produces:
+  - application/json
+
+tags:
+- name: "tweets"
+
+paths:
+
+  /tweets:
+
+    get:
+      tags:
+      - tweets
+      description: "search for a user's tweets"
+      parameters:
+      - in: "query"
+        name: userName
+        required: false
+        type: string
+
+      - in: "query"
+        name: id
+        required: false
+        type: string
+
+      responses:
+        200:
+          description: Successful response
+          schema:
+            $ref: "#/definitions/userTweets"
+        400:
+          description: The specified user ID is invalid
+          schema:
+            $ref: "#/definitions/Error"
+        500:
+          description: Unexpected Internal error
+          schema:
+            $ref: "#/definitions/Error"
+
+  /tweets/{id}:
+    get:
+      tags:
+        - tweets
+      description: "Find a tweet by tweet ID"
+      parameters:
+        - name: id
+          in: path
+          required: true
+          type: string
+
+      responses:
+        200:
+          description: Successful response
+          schema:
+            $ref: "#/definitions/tweetsData"
+        400:
+          description: Invalid request
+          schema:
+            $ref: "#/definitions/Error"
+        404:
+          description: tweet does not exist
+          schema:
+            $ref: "#/definitions/Error"
+        500:
+          description: Unexpected server error
+          schema:
+            $ref: "#/definitions/Error"
+
+    post:
+      tags:
+      - tweets
+      description: "post a new tweet"
+      parameters:
+        - name: body
+          in: body
+          description: "tweet will be created and posted"
+          required: true
+          schema:
+            $ref: "#/definitions/tweetsData"
+        - name: id
+          in: path
+          required: true
+          type: string
+
+
+      responses:
+        201:
+          description: Successfully posted
+          schema:
+            $ref: "#/definitions/tweetsData"
+        400:
+          description: Invalid syntax in request
+          schema:
+            $ref: "#/definitions/Error"
+        500:
+          description: Internal server error
+          schema:
+            $ref: "#/definitions/Error"
+
+    put:
+      tags:
+      - tweets
+      description: "update a tweet"
+      parameters:
+        - name: id
+          in: path
+          description: "update the tweet"
+          required: true
+          type: string
```

Comment:
this `id` parameter object has been defined multiple times, can you define it under [parameter object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#swagger-object) and make it reusable?